database.go

```diff
@@ -0,0 +1,1235 @@
+	return databaseSchedulesResourcePath + "/" + scheduleID + databaseBackupsResourcePath
+}
+
+// Au Resource Path
+func (au *databaseAutoscaling) resourcePath(resourceID string) string {
+	return databaseInstancesResourcePath + "/" + resourceID + databaseAutoscalingResourcePath
+}
+
+// fw Resource Path
+func (fw *databaseFirewall) resourcePath(nodeID string) string {
+	return databaseNodesResourcePath + "/" + nodeID + databaseTrustedSourcesResourcePath
+}
+
+// ta Resource Path
+func (ta *databaseTask) resourcePath(taskID string) string {
+	return databaseTasksResourcePath + "/" + taskID + "status"
+}
+
+// DatabaseListOption contains option when list resource
+type DatabaseListOption struct {
+	Page           int `json:"page,omitempty"`
+	ResultsPerPage int `json:"results_per_page,omitempty"`
+	Name           string `json:"name,omitempty"`
+}
+
+// DatabaseAutoscalingAction wrap information about autoscaling of database instance.
+type DatabaseAutoscalingAction struct {
+	Autoscaling DatabaseAutoscaling `json:"autoscaling"`
+}
+
+//DatabaseAutoscalingVolume contains volume information of autoscaling.
+type DatabaseAutoscalingVolume struct {
+	Threshold int `json:"threshold"`
+	Limited   int `json:"limited"`
+}
+
+//DatabaseAutoscaling contains autoscaling information of a database instance.
+type DatabaseAutoscaling struct {
+	Enable    bool                      `json:"enable"`
+	Volume    DatabaseAutoscalingVolume `json:"volume"`
+	Receivers []map[string]string       `json:"receivers,omitempty"`
+	Alarms    []map[string]string       `json:"alarms,omitempty"`
+}
+
+//DatabaseDNS contains DNS information.
+type DatabaseDNS struct {
+	Private string `json:"private"`
+	Public  string `json:"public"`
+}
+
+//DatabaseAddressesPrivate contains private addresses information of database.
+type DatabaseAddressesPrivate struct {
+	IPAddress   string `json:"ip_address"`
+	NetworkName string `json:"network_name"`
+}
+
+//DatabaseAddressesPublic contains public addresses information of database.
+type DatabaseAddressesPublic struct {
+	IPAddress   string `json:"ip_address"`
+	NetworkName string `json:"network_name"`
+}
+
+//DatabaseAddresses contains addresses information of database.
+type DatabaseAddresses struct {
+	Private []DatabaseAddressesPrivate `json:"private"`
+	Public  []DatabaseAddressesPublic  `json:"public"`
+}
+
+//DatabaseDatastore contains datastore information.
+type DatabaseDatastore struct {
+	Type string `json:"type"`
+	ID   string `json:"id,omitempty"`
+	Name string `json:"name,omitempty"`
+}
+
+// DatabaseAction contains database instance action information.
+type DatabaseAction struct {
+	Action     string `json:"action"`
+	NewSize    int    `json:"new_size,omitempty"`
+	FlavorName string `json:"flavor_name,omitempty"`
+	ResizeAll  bool   `json:"resize_all,omitempty"`
+	ActionAll  bool   `json:"action_all,omitempty"`
+}
+
+// DatabaseDelete contains database delete information.
+type DatabaseDelete struct {
```

Comment:
what is database object? Remember, we jus have: instances, nodes, ... but haven't database.